DaBigBlob/ani-cli
# feat: v3 (#727)

* refactor: extracted the UI into a separate file
feat: updating UI with main program
feat: announcing UI version and theme with program version

* chore: shellcheck -x to follow the sourced file

* chore: whitespaces

* feat:changed source to animixplay

* fix: shell check

* fix: always include the UI relative to the script

* chore: tyding up

* chore: formatting

* change: better interface for menus

* feat:add allanime server parsing

* choke: shellcheck

* fix: history parsing

* choke: shellcheck uwu

* feat: exclude jq dependency

* fix:optimizations

* fix:excluding extra episode

* feat: debug option, more verbose script

* fix: optimizations

* fix: shellcheck compliance for UI
improvemet: made prompt more polished

* fix: unset `$result_links` even if debug doesn't exist

* feat:argument to use gogoanime direct links only..

* qa: removed unnecessary variable: IFS

* refactor: moved `update_ui` and `ui_version_text`

* qa: broke up long lines

* qa: optimized and clarified provider selection

* qa: removed unnecessary variable `video_links`

* fix: fetch providers with indexes lower than the default

* choke: call to nonexistent function

* qa: trimmed down `append`

* "version bump" :)

* qa: removal of boilerplate

* feat:animixplay direct links provider

* feat:animixplay direct links provider

* choke:shellcheck

* docs: android referrer and pkg up (#739)

* choke: android referrer

* docs: termux pkg up (#737)

Fix for https://github.com/pystardust/ani-cli/issues/735

Co-authored-by: ivan <44473782+iamchokerman@users.noreply.github.com>
Co-authored-by: C0ff33W1thB4c0n <71843708+C0ff33W1thB4c0n@users.noreply.github.com>

* docs: android only audio

* feat:I don't know what to say

* feat: iina as MacOS default (#742)

* set iina as default for MacOS instead of mpv

* little doc update [that i accidentally removed 💀]

* ostype -> uname [posix compliance]

* readme conflict

* resolve readme conflict

* small .md err

* small .md update

* added uninsatll commands for linux & osx

* nice-to-have:preamble to the end
refactor: started to factor out the player functionality

* m3u8 resolution parsing

* refactor: mpv implementation of `play_episode`

* refactor: iina implementation of `play_episode`

* refactor: syncplay implementation of `play_episode`

* refactor: vlc implementation of `play_episode`

* screwed up the merge, backing up...

* feat:fast and asynchronous m3u8 downloading

* choke: shellcheck

* refactor: better dep_ch
feature: `player_download` script

* refactor: downloading is done by `player_download`
refactor: moved download related setup to `player_download`
fix: conforming with the new m3u8 parsing
refactor: playing/downloading messages moved to player scripts

* fix: updated download at it's place

* qa: removed preamble from daughter scripts

* qa: universal updater for every file

* choke: bad regex, files moved

* fix: shellcheck
note: export in `player_download` is only required by shellcheck because it doesn't check conditionally sourced files

* qa: spaces and other elegance things
qa: identifying comments to the m3u8 download scheme

* qa: shorter m3u8 quality selection
fix: if set quality isn't found, only the current ep will play on best

* fix: shellcheck (pls for the last time)

* fix:no video_url in m3u8

* fix:background jobs checking

* qa:select provider, refactoring

* qa: whitespaces and useless ocd changes

Co-authored-by: CoolnsX <ansari.tan20@gmail.com>
Co-authored-by: chokerman <44473782+justchokingaround@users.noreply.github.com>
Co-authored-by: ivan <44473782+iamchokerman@users.noreply.github.com>
Co-authored-by: C0ff33W1thB4c0n <71843708+C0ff33W1thB4c0n@users.noreply.github.com>
Co-authored-by: Person <73036332+DaBigBlob@users.noreply.github.com>

# Pull Request Template

## Type of change


## Description

*ramble here*

## Checklist

- [ ] any anime playing
- [ ] next, prev and replay work
- [ ] quality works
- [ ] downloads work
- [ ] quality works with downloads
- [ ] select episode -a and rapid resume work
- [ ] syncplay -s works
- [ ] autoplay, aka range selection, works

## Additional Testcases

- The safe bet: One Piece
- Episode 0: Saenai Heroine no Sodatekata ♭
- Unicode: Saenai Heroine no Sodatekata ♭
- Not uploaded: one piece dub episode 590
- Unreleased: soredemo ayumu wa yosetekuru
- Short id (for decryption): Log Horizon episode 1-2
